mule-artifact-it/mule-packager-it/src/test/java/integration/test/mojo/AdditionalPluginDependenciesProcessSourcesMojoTest.java

```diff
@@ -140,6 +140,7 @@ public void muleAppWithReusableAppPlugin() throws Exception {
    * plugins.
    */
   @Test
+  // TODO: Lucho could you explain what we are testing here?
```

Comment:
We should be testing that the classloader-model.json is correctly generated and the additionalPluginDependencies added the expected ones. 
Now, in this particular case, i actually don't how the precedence should be. It seems that if an app is reusing another app that declares an additionalPluginDependency in one of it's plugins. And the first app does the same with one of that app's plugins, then the higher declaration will take place.